src/app/routes/routeLogin.ts

```diff
@@ -0,0 +1,85 @@
+import { Router } from "express";
+import { z } from "zod";
+import jwt from "jsonwebtoken";
+import createError from "http-errors";
+
+import { parseRequest } from "../middlewares/parseRequest";
+import { handleAsync } from "../utils/handleAsync";
+import { container } from "../../domains/@config/di.config";
+import { UserService } from "../../domains/userService";
+import { OauthClient } from "../../infrastructure/github";
+import { Github } from "../../infrastructure/github";
+import Config from "../../config";
+
+export const loginRouter = Router();
+
+loginRouter.get(
+  "/login",
+  parseRequest({
+    query: z.object({
+      code: z.string(),
+    }),
+  }),
+  handleAsync(async (req, res, next) => {
+    const { username, userGithubUri, userImage } = req.app.locals;
+    const { code } = req.query;
+    const oauthClient = new OauthClient();
+    const githubAccessToken = await oauthClient.getToken(code);
+
+    if (!githubAccessToken) {
+      return next(
+        createError(
+          401,
+          "Authentication failed. Cannot find 'githubAccessToken'.",
+        ),
+      );
+    }
+
+    const github = new Github(githubAccessToken);
+    const githubData = await github.getUserData();
+
+    if (!githubData) {
+      return next(
+        createError(401, "Authentication failed. Cannot find 'githubData'."),
+      );
+    }
+
+    const userService = container.get<UserService>("UserService");
+
+    const userData = await userService.login({
+      username,
+      userGithubUri,
+      userImage,
+      githubAccessToken,
+    });
+
+    const userPayload = {
+      username: userData?.username,
+      userGithubUri: userData?.userGithubUri,
+      userImage: userData?.userImage,
+    };
+
+    const accessToken = jwt.sign(userPayload, Config.JWT_SECRET, {
+      expiresIn: "1d",
+    });
+
+    // TODO - cookie 수정
+    const { referer } = req.headers;
+    const loginData = JSON.stringify({
+      id: userData?._id,
+      name: userData?.username,
+      githubUri: userData?.userGithubUri,
+      image: userData?.userImage,
+      githubAccessToken,
+      accessToken,
+    });
+
+    return res
```

Comment:
쿠키 옵션으로는

```suggestion
  const httpsOptions = Config.NodeEnv === 'production'
      ? {
          secure: true,
        }
      : {};
  }

  const cookieOptions = {
    expires: new Date(Date.now() + 24 * 60 * 60 * 1000),
    maxAge: 24 * 60 * 60 * 1000,
    sameSite: 'lax',
    ...httpsOption,
  }

  const loginData = JSON.stringify({
    id: userData?._id,
    name: userData?.username,
    githubUri: userData?.userGithubUri,
    image: userData?.userImage,
  });

  res.cookie('githubAccessToken', githubAccessToken, cookieOptions);
  res.cookie('accessToken', accessToken, cookieOptions);
  res.cookie('loginData', githubAccessToken, cookieOptions);
```

이렇게 해보는 건 어떨까요? 프론트로 github 로직을 옮기게 되면 loginData부분만 없애면 될 것 같아용

Reply:
request header에 cookie가 자동으로 들어오지 않는건.. 이유를 모르겠네요.. withCredential: true로도 해봤는데 안되네요..ㅎ

Reply:
오 넵 이거는 다음 PR 때 바로 적용해서 올리겠습니당! 셋 다 넣어주는게 아무래도 나중에 변경하기 편하겠네요